sdk/metric/exemplar.go

```diff
@@ -69,7 +69,11 @@ func DefaultExemplarReservoirProviderSelector(agg Aggregation) exemplar.Reservoi
 		//   provided, the default size MAY be the number of possible
 		//   concurrent threads (e.g. number of CPUs) to help reduce
 		//   contention. Otherwise, a default size of 1 SHOULD be used.
-		n = runtime.NumCPU()
+		//
+		// Use runtime.GOMAXPROCS instead of runtime.NumCPU to support
+		// containerized environments that may have less than the total number
+		// of logical CPUs available on the local machine allocated to it.
+		n = runtime.GOMAXPROCS(0)
```

Comment:
Are there ever cases where people set GOMAXPROCS higher than NumCPU, and would be surprised by this change?  Should we use the min of the two?